projects/mc_interface/src/main.c

```diff
@@ -24,6 +24,21 @@ static SequencedRelayStorage s_relay_storage;
 static HeartbeatRxHandlerStorage s_powertrain_heartbeat;
 static UARTStorage s_uart_storage;
 
+static void prv_check_precharge_relays(SoftTimerID timer_id, void *context) {
+  SequencedRelaySettings *relay_settings = (SequencedRelaySettings *)context;
+
+  GPIOState precharge_left, precharge_right;
+
+  gpio_get_state(&relay_settings->left_sense, &precharge_left);
+  gpio_get_state(&relay_settings->right_sense, &precharge_right);
+
+  if ((precharge_left == GPIO_STATE_LOW) || (precharge_right == GPIO_STATE_LOW)) {
+    sequenced_relay_init(&s_relay_storage, relay_settings);
+    soft_timer_start_millis(MC_CFG_PRECHARGE_CHECK_INTERVAL_MS, prv_check_precharge_relays,
```

Comment:
Instead of checking periodically if this failed you should reset the precharge relay right and left control pins instead. i.e. pretend another relay close event was triggered and attempt to close the relays again.